lib/morse_translator.rb

```diff
@@ -40,4 +41,50 @@ def initialize
                     " " => " "}
   end
 
+  def translate(message)
+    translation = ""
+    find_morse_code_words(message).each do |code|
+      translation += code
+    end
+    return translation
+  end
+
+  def find_morse_code_words(message)
+      letters = message.downcase.chars
+      letters.map do |letter|
+        find_morse_code_letter(letter)
+      end
+  end
+
+  def find_morse_code_letter(character)
+    dictionary.find do |letter|
```

Comment:
I LOVE that you are using the find method on a hash!! So great.  Just a couple of things to think about:

1. When you iterate over a hash, you can send two variables into the block so that you have a 'key' and a 'value' variable.  For example, you could do something like this:

```ruby
@dictionary.find do |letter, morse|
  if letter == character
    return morse
  end
end
```

2. Also, don't forget that if you know the key of a hash you can look up the value associated with that key directly (you don't have to go 'finding' for it).  You could do that like this:

``` ruby
def find_morse_code_letter(character)
  @dictionary[character]
end
```